src/test/py/peer/peer_test.py

```diff
@@ -3,72 +3,120 @@
+        """Stores a value based on the key."""
+        self.data_store[key] = value
+
+    def get(self, key):
+        """Returns a value based on the key."""
+        if key in self.data_store:
+            return self.data_store[key]
+        raise KeyError(f"No value found for key: {key}")
+
+    async def bootstrap(self, peer):
+        """Bootstrap the node with a peer."""
+        if len(peer) > 0:
+            return True
+        return False
+
+    async def listen(self, port):
+        """Listen on the port."""
+        if port > 0:
+            return True
+        return False
+
+
+class TestPeer(unittest.IsolatedAsyncioTestCase):
     """
     Class to test peer functionality.
     """
 
+    test_logger = logging.getLogger("TestPeer")
+
     def setUp(self):
         """
         Initialize IPFS Mock and Peer instance
         """
-        self.mock_ipfs = Mock()
-        self.peer = Peer("127.0.0.1", 5001, self.mock_ipfs)
+
+        self.mock_kdm = MagicMock()
+        self.mock_node = AsyncMock(spec=MockNode)
+        self.mock_kdm.network.Server.return_value = self.mock_node
+        self.peer = Peer(5001, "127.0.0.1:5000", self.mock_kdm)
 
     def test_initialization(self):
         """
         Test case to verify the initialization of the Peer class.
         """
         # Verify that the network address is set to "127.0.0.1"
-        self.assertEqual(self.peer.network_address, "127.0.0.1")
+        self.assertEqual(str(self.peer.network_ip_address), "127.0.0.1")
         # Verify that the port is set to 5001
         self.assertEqual(self.peer.port, 5001)
-        # Verify that the IPFS instance is set to the mock_ipfs object
-        self.assertEqual(self.peer.ipfs, self.mock_ipfs)
+        # Verify that the IPFS instance is set to the mock_kdm object
+        self.assertEqual(self.peer.kdm, self.mock_kdm)
 
-    def test_connect(self):
+    async def test_connect(self):
         """
         Test case for connecting to the network.
         """
-        self.mock_ipfs.connect.return_value = True
-        result = self.peer.connect_to_network()
-        self.assertTrue(result)
-        self.mock_ipfs.connect.assert_called_once()
+        await self.peer.connect_to_network()
+        self.peer.node.bootstrap.assert_called_once()
 
     @patch("builtins.input", side_effect=["10", "20", "10"])
     @patch(
         "threading.Timer",
-        side_effect=lambda delay, func, args: Timer(0.001, func, args),
+        side_effect=lambda delay, func, args: Timer(0, func, args),
     )
     def test_commission_art_piece(self, mock_input, mock_timer):
         """
         Test case for the commission_art_piece method of the Peer class.
         This test verifies that the commission_art_piece method correctly adds a commission,
         publishes it on IPFS, and schedules and sends a deadline notice.
         """
-        print(mock_input, mock_timer)
+        self.test_logger.info(mock_input, mock_timer)
```

Comment:
`debug` level would be better IMO if this is needed. But, I'm curious, after seeing what this prints out, it doesn't seem that useful. What am I missing?